src/main/features/certificateOfService/routes/file-upload.ts

```diff
@@ -17,29 +19,47 @@ export default express.Router()
 
       form.parse(req)
       .on('file', function (name, file) {
-        if (file.size === 0) {
-          res.redirect(Paths.documentUploadPage.uri)
-        }
         const user: User = res.locals.user
-        DocumentsClient.save(user.bearerToken, file).then((documentManagementURI) => {
-          let files: UploadedDocument[] = []
-          const documentType: DocumentType = user.legalUploadDocumentDraft.document.fileToUpload
-          user.legalCertificateOfServiceDraft.document.uploadedDocuments.map((file) => files.push(new UploadedDocument().deserialize(file)))
-          files.push(new UploadedDocument(file.name, file.type, documentType, documentManagementURI))
-
-          fs.unlink(file.path, function (err) {
-            if (err) {
-              next(err)
-            }
+        if (file.size === 0) {
+          user.legalUploadDocumentDraft.document.fileToUploadError = FileTypeErrors.FILE_REQUIRED
+          new DraftService().save(user.legalUploadDocumentDraft, user.bearerToken).then(() => {
```

Comment:
This seems duplicated in all blocks, can we please extract this out

Reply:
done